### PR TITLE
Optimized some miscellaneous code

### PR DESCRIPTION
**Pull request by ghost**
Just a few things clippy picked up that seemed like easy fixes

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! Client library for the [Discord](https://discordapp.com) API.
 //!
-//! The Discord API can be divided into three main components: the RESTful API
+//! The Discord API can be divided into three main components: the `RESTful` API
```
Comment (ghost) [inline src/lib.rs]: Clippy is often incorrect here. It's meant to notify about structs/modules which should be ticked for highlighting, but in this case un-ticked is correct.

Relevant lint: https://github.com/Manishearth/rust-clippy/wiki#doc_markdown

File: src/model.rs [hunk 2]
```diff
@@ -1391,7 +1390,7 @@ impl RelationshipType {
 	}
 
 	fn decode(value: Value) -> Result<Self> {
-		value.as_u64().and_then(RelationshipType::from_num).ok_or(Error::Decode("Expected valid RelationshipType", value))
+		value.as_u64().and_then(RelationshipType::from_num).ok_or_else(|| Error::Decode("Expected valid RelationshipType", value))
```
Comment (ghost) [inline src/model.rs]: `ok_or_else` over `ok_or` is not necessary _in this case_ as it doesn't make any functional difference (and is probably less performant? unsure if that's optimized away), but perhaps Space likes it better.

Comment (ghost) [reply src/model.rs]: See https://github.com/Manishearth/rust-clippy/wiki#or_fun_call for why `ok_or_else` is preferred, although I'm not sure it's needed in this example.

File: src/lib.rs (hunk 1 above)
Comment (ghost) [reply src/lib.rs]: Fixed, sorry about that.

File: src/model.rs (hunk 2 above)
Comment (ghost) [reply src/model.rs]: Alright, change reverted since it doesn't make any difference in this case.

**Comment by SpaceManiac** [comment]
Looks pretty good. Just three nitpicks this way around:
- The `field_name: _` to `..` changes I would like to be excluded. The explicitness helps with keeping these matches up to date when the types change.
- The `|` in the match in `error.rs` should be split between lines:

``` rust
Error::Decode(msg, _) |
Error::Protocol(msg) |
Error::Other(msg) => msg,
```
- The `match` to `if let` changes should place the whole condition on one line:

``` rust
if let Some(x) = y.iter_mut().find(...) {
    ...
}
```

File: src/model.rs [hunk 3]
```diff
@@ -2077,7 +2077,7 @@ impl VoiceEvent {
 // Decode helpers
 
 fn remove(map: &mut BTreeMap<String, Value>, key: &str) -> Result<Value> {
-	map.remove(key).ok_or(Error::Decode("Unexpected absent key", Value::String(key.into())))
+	map.remove(key).ok_or_else(|| Error::Decode("Unexpected absent key", Value::String(key.into())))
```
Comment (ghost) [inline src/model.rs:2080]: This `ok_or` conversion to `ok_or_else` is probably not necessary here as well.

File: src/state.rs [hunk 4]
```diff
@@ -133,10 +133,9 @@ impl State {
 				if let Some(channel) = state.channel_id {
 					// channel id available, insert voice state
 					if let Some(call) = self.calls.get_mut(&channel) {
-						match call.voice_states.iter_mut().find(|u| u.user_id == state.user_id) {
-							Some(grp_state) => { grp_state.clone_from(state); return }
-							None => {},
-						}
+						if let Some(grp_state) = call.voice_states.iter_mut().find(|u| u.user_id == state.user_id) {
+								grp_state.clone_from(state); return
+							}
```
Comment (ghost) [inline src/state.rs]: Looks like the indentation got a little messed up here and the next diff block, as well as 3 blocks down.

File: src/model.rs (hunk 3 above)
Comment (SpaceManiac) [reply src/model.rs:2080]: This one is okay to keep. Saves on allocating a `String` in the success case.

**Comment by SpaceManiac** [comment]
Merged in 6ff0f21e58821d35c6b120a6ef4890f31d6b757a. Thanks.

**Pull request closed without merging**